codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2RemoteFlowController.java

```diff
@@ -277,24 +282,27 @@ private int allocateBytesForTree(Http2Stream parent, int connectionWindow) {
 
         // This is the priority algorithm which will divide the available bytes based
         // upon stream weight relative to its peers
-        Http2Stream[] children = parent.children().toArray(new Http2Stream[parent.numChildren()]);
+        ChildCache childCache = new ChildCache(parent);
         int totalWeight = parent.totalChildWeights();
-        for (int tail = children.length; tail > 0;) {
-            int head = 0;
-            int nextTail = 0;
+        while (childCache.nextIterationSize() > 0) {
             int nextTotalWeight = 0;
             int nextConnectionWindow = connectionWindow;
-            for (; head < tail && nextConnectionWindow > 0; ++head) {
-                Http2Stream child = children[head];
+
+            for (Http2Stream child : childCache) {
```

Comment:
I wonder if we can avoid creating new iterator objects each time...seems like we have the potential to increase GC a bit here.